apps/arweave/src/ar_util.erl

```diff
@@ -124,11 +124,12 @@ parse_peer("") -> throw(empty_peer_string);
 parse_peer(BitStr) when is_bitstring(BitStr) ->
 	parse_peer(bitstring_to_list(BitStr));
 parse_peer(Str) when is_list(Str) ->
-	case io_lib:fread("~d.~d.~d.~d", Str) of
-		{ok, [A, B, C, D], PortStr} ->
+    [Addr, PortStr] = parse_port_split(Str),
+    case inet:getaddr(Addr, inet) of
+		{ok, {A, B, C, D}} ->
 			{A, B, C, D, parse_port(PortStr)};
-		{error, _} ->
-			throw({invalid_peer_string, Str})
+		{error, Posix} ->
```

Comment:
Why "Posix"?

Reply:
it's a posix return error code, i just took the name from the erlang lib description

Reply:
`posix()` is a spec for names(atoms) of POSIX error codes. More better name can be `Reason` or you can skip this `Reason` as was before